WordPress/Classes/ViewRelated/Post/PostSettingsViewController.m

```diff
@@ -532,7 +547,10 @@ - (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPa
     }
 
     if (sectionId == PostSettingsSectionMeta) {
-        if (indexPath.row == RowIndexForPassword) {
+        if (indexPath.row == PostMetaRowAuthor && ![self canShowAuthorSelection]) {
+            return 0;
+        }
+        if (indexPath.row == PostMetaRowPassword) {
```

Comment:
When a post shouldn't show authors, is the author cell still present in the table view data source but is rendered with a height of zero?

Reply:
Correct, it's also marked as `hidden` [here](https://github.com/wordpress-mobile/WordPress-iOS/pull/16281/files#diff-37b18ea4939f28d52d0e63f7a5fa45de61fe0ad85c3d402e98b6e5e10a1dc45cR668).

I don't find this pleasant or graceful at all, but it's what came up in my research. 😄  Will be glad to research more. My biggest concern would be a11y (we don't want it to exist if hidden), and I can double-check that.

Reply:
VoiceOver did not appear to pick up the Author button when it wasn't shown.

Reply:
I hadn't noticed that it was hidden as well, so that helps. I was wondering though, would actually not including the cell at all be possible? If a cell is not needed I would have expected it to not be in the data source at all, but that doesn't appear to be the case here. What made you prefer hiding the cell instead here?

Reply:
Originally, all rows were shown, but the Password was optional. This was easy to control because we could just +1 the rows in the meta section.

Since the Author option was at the top I took a different approach. I agree, the cell should never be created if it isn't needed. I'll come up with a better solution.

Reply:
A proposed fix has been implemented.

Reply:
Thanks! I really like the new solution. I think the approach used for the password row was not scalable and was bound to break and now just happened to be that time.